node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -477,14 +480,29 @@ setPtxCondition :: CId Wal -> TxId -> PtxCondition -> Update ()
 setPtxCondition wid txId cond =
     wsWalletInfos . ix wid . wsPendingTxs . ix txId . ptxCond .= cond
 
+-- | Conditional modifier.
+-- Returns 'True' if pending transaction existed and modification was applied.
+checkAndSmthPtx
+    :: CId Wal
+    -> TxId
+    -> (Maybe PtxCondition -> Bool)
+    -> LS.State (Maybe PendingTx) ()
+    -> Update Bool
+checkAndSmthPtx wid txId whetherModify modifier =
+    fmap getAny $ zoom' (wsWalletInfos . ix wid . wsPendingTxs . at txId) $ do
+        matches <- whetherModify . fmap _ptxCond <$> get
+        when matches modifier
+        return (Any matches)
```

Comment:
That's some dense code, but here is my interpretation: Zoom to focus on a particular `PendingTx` given its `TxId`, and if the predicate `whetherModify` matches, run `modifier`. Then return true if we had a match in the first place.

A minor nitpick would be that this function is really not enforcing that `modifier` actually _did_ something, just that we had a match in the first place, which is probably fine(?).